packages/babel-parser/src/parser/statement.js

```diff
@@ -2042,6 +2061,52 @@ export default class StatementParser extends ExpressionParser {
     node.specifiers.push(this.finishNode(specifier, type));
   }
 
+  maybeParseModuleAttributes() {
+    if (!this.eat(tt._with)) {
+      if (this.hasPlugin("moduleAttributes")) return [];
+      return null;
+    }
+    this.expectPlugin("moduleAttributes");
+    const attrs = [];
+    const attributeMap = {};
+    do {
+      // we are trying to parse a node which has the following syntax
+      // with type: "json"
+      // [with -> keyword], [type -> Identifier], [":" -> token for colon], ["json" -> StringLiteral]
+      const node = this.startNode();
+      node.key = this.parseIdentifier(true);
+
+      // check if we already have an entry for an attribute
+      // if a duplicate entry found, throw an error
+      if (attributeMap[node.key.name]) {
+        throw this.raise(
+          this.state.start,
+          Errors.ModuleAttributesWithDuplicateKeys,
+        );
+      }
+      // set the attribute name entry in the attributeMap to true
+      attributeMap[node.key.name] = true;
+      // check for colon
+      this.expect(tt.colon);
+      // check if the value set to the module attribute is a string as we only allow string literals
+      if (!this.match(tt.string)) {
+        return this.unexpected(
+          this.state.start,
+          "Only string literal values are allowed",
```

Comment:
Please move this to the `Errors` object.

Also, let's give context to the error message:
```suggestion
          "Only string literals are allowed as module attributes values",
```